pacman/model/graphs/application/application_graph.py

```diff
@@ -18,44 +18,155 @@
 from .application_vertex import ApplicationVertex
 from .application_edge_partition import ApplicationEdgePartition
 from spinn_utilities.ordered_set import OrderedSet
-from spinn_utilities.overrides import overrides
 from pacman.exceptions import (
     PacmanAlreadyExistsException, PacmanInvalidParameterException)
-from pacman.model.graphs.graph import Graph
 
 
-class ApplicationGraph(Graph):
+class ApplicationGraph(object):
     """ An application-level abstraction of a graph.
     """
 
     __slots__ = [
         # The sets of edge partitions by pre-vertex
         "_outgoing_edge_partitions_by_pre_vertex",
         # The total number of outgoing edge partitions
-        "_n_outgoing_edge_partitions"
+        "_n_outgoing_edge_partitions",
+        # count of vertex which had a None or already used label
+        "_unlabelled_vertex_count",
+        # map between labels and vertex
+        "_vertex_by_label"
     ]
 
-    def __init__(self, label):
-        """
-        :param label: The label on the graph, or None
-        :type label: str or None
-        """
-        super().__init__(ApplicationVertex, ApplicationEdge, label)
+    def __init__(self):
         self._outgoing_edge_partitions_by_pre_vertex = defaultdict(OrderedSet)
         self._n_outgoing_edge_partitions = 0
+        self._unlabelled_vertex_count = 0
+        self._vertex_by_label = dict()
+
+    def add_vertex(self, vertex):
+        """ Add a vertex to the graph.
+
+        :param ApplicationVertex vertex: The vertex to add
+        :raises PacmanInvalidParameterException:
+            If the vertex is not of a valid type
+        :raises PacmanConfigurationException:
+            If there is an attempt to add the same vertex more than once
+        """
+        if not isinstance(vertex, ApplicationVertex):
+            raise PacmanInvalidParameterException(
+                "vertex", str(vertex.__class__),
+                "Only ApplicationVertexs can be added")
+        if not vertex.label:
+            vertex.set_label(
+                vertex.__class__.__name__ + "_" + self._label_postfix())
+        elif vertex.label in self._vertex_by_label:
+            if self._vertex_by_label[vertex.label] == vertex:
+                raise PacmanAlreadyExistsException("vertex", vertex.label)
+            vertex.set_label(vertex.label + self._label_postfix())
+        vertex.addedToGraph()
+        self._vertex_by_label[vertex.label] = vertex
+
+    @property
+    def vertices(self):
+        """ The vertices in the graph.
+
+        :rtype: iterable(AbstractVertex)
```

Comment:
Perhaps say that this is a dict rather than just an iterable?

Reply:
The call returns only the values(0 of the dict not the whole dict. So iterable is correct